support/sier-codec/src/schema.rs

```diff
@@ -146,6 +148,18 @@ impl Type {
 
         match self {
             Type::Unit => Ok((bytes, Value::Unit)),
+            Type::Bool => {
+                let (b, n) = complete::le_u8(bytes).map_err(Error::ValueParsing)?;
+                let value = match n {
+                    0 => Ok(Value::Bool(false)),
+                    1 => Ok(Value::Bool(true)),
+                    _ => Err(Error::ValueParsing(nom::Err::Error(
+                        nom::error::make_error(bytes, nom::error::ErrorKind::IsNot),
+                    ))),
```

Comment:
I'd have the match statement just be true/false and the error case explicitly return.

```rust
let value = match n {
  0 => false,
  1 => true,
  _ => return Err(...),
};
Ok((b, Value::Bool(value)))
```